arm-network/2017-03-01/swagger/networkWatcher.json

```diff
@@ -1714,6 +1769,176 @@
           "$ref": "#/definitions/FlowLogProperties"
         }
       }
+    },
+    "ConnectivityParameters": {
+      "description": "Parameters that determine how the connectivity check will be performed.",
+      "required": [
+        "source",
+        "destination"
+      ],
+      "properties": {
+        "source": {
+          "$ref": "#/definitions/ConnectivitySource"
+        },
+        "destination": {
+          "$ref": "#/definitions/ConnectivityDestination"
+        }
+      }
+    },
+    "ConnectivitySource": {
+      "description": "Parameters that define the source of the connection.",
+      "required": [
+        "resourceId"
+      ],
+      "properties": {
+        "resourceId": {
+          "description": "The ID of the resource from which a connectivity check will be initiated.",
+          "type": "string"
+        },
+        "port": {
+          "description": "The source port from which a connectivity check will be performed.",
+          "type": "integer"
+        }
+      }
+    },
+    "ConnectivityDestination": {
+      "description": "Parameters that define destination of connection.",
+      "properties": {
+        "resourceId": {
+          "description": "The ID of the resource to which a connection attempt will be made.",
+          "type": "string"
+        },
+        "address": {
+          "description": "The IP address or URI the resource to which a connection attempt will be made.",
+          "type": "string"
+        },
+        "port": {
+          "description": "Port on which check connectivity will be performed.",
+          "type": "integer"
+        }
+      }
+    },
+    "ConnectivityInformation": {
+      "description": "Information on the connectivity status.",
+      "properties": {
+        "hops": {
+          "type": "array",
+          "description": "List of hops on between the source and the destination.",
+          "items": {
+            "$ref": "#/definitions/ConnectivityHop"
+          }
+        },
+        "connectionStatus": {
+          "type": "string",
+          "enum": [ "Unknown", "Connected", "Disconnected", "Degraded" ],
+          "x-ms-enum": {
+            "name": "ConnectionStatus",
+            "modelAsString": true
+          },
+          "description": "The connection status."
+        },
+        "avgLatencyInMs": {
+          "description": "Average latency in milliseconds.",
```

Comment:
is this property (or the following properties) *always* provided by your service? If so, consider marking it as `"x-nullable": false` in order to generate non-nullable value types (e.g. will generate `int` instead of `int?` in C#).